### ISSUE TITLE
Force correct URL format for custom URL on profile edit page

### ISSUE THREAD

**Issue by skylenewman** [OpenHuntsville/hntsvll#68]
<img width="814" alt="screen shot 2016-07-30 at 3 41 41 pm" src="https://cloud.githubusercontent.com/assets/1457762/17273012/5efc5eb4-566c-11e6-9647-7b35f1095b55.png">

The attached screenshot contains characters that I should not be able to add. Users should only be able to enter alphanumeric characters and dashes

Strings should be validated via JS on the front end, as well as the Ruby back end.

**Comment by RandomlyKnighted** [comment]
What characters should be accepted?

**Comment by davidhjones** [comment]
@RandomlyKnighted This is very closely related to #128.

**Comment by RandomlyKnighted** [comment]
Hmm forgot about this issue. 

Anyways, @skylenewman @davidhjones I think the best way to handle this issue as well as #128 and   #102 would be to add a polyfill for HTML5 so that attributes like required and pattern work with Safari and other browsers that don't support it. This way we don't have to add custom code to get around any attribute we may we want to use. 

What do you think?

**Comment by davidhjones** [comment]
I don't have experience with Polyfill but that makes sense. If you go down this route, make sure to remove the custom code that we have on the `events/_form.html` to handle required tags on Safari. Also, we really should be adding code on the Ruby side too. I can help you with this, but I think it should be included in these issues.

**Comment by RandomlyKnighted** [comment]
Yeah that's fine. I need to look into Ring.js so I can get a better grasp on how server side validation would affect any work I'm doing on the front-end.

**Comment by davidhjones** [comment]
I don't think you'd have to leverage Ring.js. I think you could simply in the object creation (in the routing `:create` and `:update` function) add logic that verifies the inputs are what expect).

**Comment by RandomlyKnighted** [comment]
Well by that point you've already done a submit on the page and are in the process of being redirected are you not? I was thinking it'd be nice if we could verify on the back end and then update the front end to let the user know that something is invalid.

**Comment by davidhjones** [comment]
Oh I see what you're thinking. I'm thinking the UI validation should be done on the front-end, and for security reason we should validate on the backend. Technically, if we don't have UI front-end validation, someone could directly ping the endpoints. Therefore I was thinking we should be checking that the information we're receiving in a POST or UPDATE is actually valid data

**Comment by davidhjones** [comment]
Although, to your point with Ring.js, I do think that is a possible solution (doing all the validation in Ruby) if you want to go that route for UI interaction.

**Comment by RandomlyKnighted** [comment]
I ran across this jQuery polyfill for HTMl5 validation: https://github.com/ericelliott/h5Validate
I'll be using it to fix these issues.

**Note:** Looking at caniuse.com it looks like Safari will have full support for the required and pattern attributes with version 10.1. I did look through Apple's version history for Safari and they stay pretty consistent with releases. If they follow their previous patterns then 10.1 should be released within the next couple of months. If so then we can always revisit this issue later in the year and see if a polyfill is still needed.

On a similar note the only other HTML5 attribute I can think of that I commonly use is the placeholder attribute and it's supported by Safari just not IE. If I remember right that can be polyfilled using html5shiv, but I'll look into that when we actually need it.
